packages/url/url_common.js

```diff
@@ -1,21 +1,28 @@
 URL = {};
 
-var encodeString = function(str) {
-  return encodeURIComponent(str).replace(/[!'()]/g, escape).replace(/\*/g, "%2A");
+var encodeString = function (str) {
+  return encodeURIComponent(str).replace(/\*/g, '%2A');
 };
 
-
-URL._encodeParams = function(params) {
-  var buf = [];
-  _.each(params, function(value, key) {
-    if (buf.length)
-      buf.push('&');
-    buf.push(encodeString(key), '=', encodeString(value));
-  });
-  return buf.join('').replace(/%20/g, '+');
+// Encode URL paramaters into a query string, handling nested objects and
+// arrays properly.
+URL._encodeParams = function (params, prefix) {
+  var str = [];
+  for (var p in params) {
+    if (params.hasOwnProperty(p)) {
```

Comment:
I know this may seem like an edge case, but what if there's a parameter named `hasOwnProperty`?